iot-hub/Samples/device/PnpDeviceSamples/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -107,6 +126,47 @@ public async Task PerformOperationsAsync(CancellationToken cancellationToken)
             }
         }
 
+        private async Task GetWritablePropertiesAndHandleChangesAsync()
+        {
+            Twin twin = await _deviceClient.GetTwinAsync();
+            _logger.LogInformation($"Device retrieving twin values on CONNECT: {twin.ToJson()}");
+
+            TwinCollection twinCollection = twin.Properties.Desired;
+            long serverWritablePropertiesVersion = twinCollection.Version;
+
+            // Check if the writable property version is outdated on the local side.
+            // For the purpose of this sample, we'll only check the writable property versions between local and server
+            // side without comparing the property values.
+            if (serverWritablePropertiesVersion > s_localWritablePropertiesVersion)
+            {
+                _logger.LogInformation($"The writable property version cached on local is changing " +
+                    $"from {s_localWritablePropertiesVersion} to {serverWritablePropertiesVersion}.");
+
+                foreach (KeyValuePair<string, object> propertyUpdate in twinCollection)
+                {
+                    string componentName = propertyUpdate.Key;
+                    switch (componentName)
+                    {
+                        case Thermostat1:
+                        case Thermostat2:
+                            if (!_temperature.TryGetValue(componentName, out double value))
+                            { 
+                                _temperature[componentName] = 0d;
```

Comment:
Maybe a better default would be 70F (or equivalent in °C if that is the unit specified in the PnP model).